pywbemtools/pywbemcli/_pywbem_server.py

```diff
@@ -418,16 +419,35 @@ def conn(self):
 
         `None` when disconnected from the server.
         """
-        return self.wbem_server.conn if self.wbem_server else None
+        if self._conn:
+            return self._conn
+
+        self._conn = self.wbem_server.conn
+        return self._conn
 
     @property
     def wbem_server(self):
         """
-        :class:`~pywbem.WBEMServer`: Server object to be used for higher level
-        WBEM server requests, when connected to the server.
+        :class:`~pywbem.WBEMServer`: WBEMServer instance to be used for
+        higher level WBEM server requests, when connected to the server.
 
         `None` when disconnected from the server.
-        """
+
+        Using this property will connect to the server if not yet connected,
+        and will raise an exception if no server is specified.
+
+        When connecting to a real WBEM server, a password is prompted for if
+        a user is specified in the current context but no password. The
+        password is saved in the context, so the password is prompted for only
+        once (e.g. in interactive mode).
+        """
+        ctx = click.get_current_context()
+        if self._wbem_server is None:
+            self.get_password(ctx.obj)
+            self.connect(
+                log=ctx.obj.log,
+                use_pull=ctx.obj.use_pull,
+                verbose=ctx.obj.verbose)
```

Comment:
The corresponding disconnect()  is missing, so we are not closing properly.

Reply:
This one confuses me.  We do have a disconnect method in PywbemServer (line 518).  It is called from two places

1. pywbemcli line718 - This is the normal exit.
2. _cmd_connection line 694 - disconnect any existing connection when a select is executed.
3. The code to execute it when the user enters an interactive command that changes the server is not in yet.

We may be missing some exit method but I thought we were complete here

**DISCUSSION**

Reply:
Ok, I did some more investigation on this. Issues are:
* The whole purpose of `disconnect()` is to call `conn.close()`. I had prepared the `PR900_DISCONNECT_CALLS_CLOSE` flag to enable that, but it is not set to `True` so `close()` is not called. It needs to be set to True.
TODO(Andy): More to come

Reply:
Resolved in the other PR.
DONE